clang/lib/Driver/Driver.cpp

```diff
@@ -5696,9 +5765,13 @@ class OffloadingActionBuilder final {
             // the following iterations.
             FoundNormalizedTriples[NormalizedName] = Val;
 
-            SYCLTripleList.push_back(TT);
+            SYCLTripleList.push_back(C.getDriver().MakeSYCLDeviceTriple(Val_t));
             if (TT.getSubArch() == llvm::Triple::SPIRSubArch_fpga)
               SYCLfpgaTriple = true;
+            // For user specified spir64_gen, add an empty device value as a
+            // placeholder.
+            if (TT.getSubArch() == llvm::Triple::SPIRSubArch_gen)
```

Comment:
I didn't see a test checking for `-fsycl-traget=spirv64_gen,intel_gpu_skl` (or similar), unless I missed it, I think it needs one.

Reply:
Additional testing added